toolkits/confluence/arcade_confluence/client.py

```diff
@@ -0,0 +1,535 @@
+from enum import Enum
+from typing import Any
+from urllib.parse import parse_qs, urlparse
+
+import httpx
+from arcade.sdk.errors import ToolExecutionError
+
+from arcade_confluence.enums import BodyFormat, PageUpdateMode
+from arcade_confluence.utils import (
+    build_child_url,
+    build_hierarchy,
+    remove_none_values,
+    split_by_space,
+)
+
+
+class ConfluenceAPIVersion(str, Enum):
+    V1 = "wiki/rest/api"
+    V2 = "wiki/api/v2"
+
+
+class ConfluenceClient:
+    ACCESSIBLE_RESOURCES_URL = "https://api.atlassian.com/oauth/token/accessible-resources"
+    BASE_URL = "https://api.atlassian.com/ex/confluence"
+
+    def __init__(self, token: str, api_version: ConfluenceAPIVersion):
+        self.token = token
+        self.cloud_id = self._get_cloud_id()
```

Comment:
Nit: ideally, I wouldn't want a class to issue an HTTP request when it's being instantiated. What I did for Jira client was smth like:

```
def __init__(self, cloud_id = None):
        self._cloud_id = cloud_id

@property
def cloud_id(self):
    if not self._cloud_id:
        self._cloud_id = self._get_cloud_id()
    return self._cloud_id
```